pkg/storage/tsdb/cacheable_index.go

```diff
@@ -0,0 +1,310 @@
+package tsdb
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
```

Comment:
We'll probably want to use an encoding scheme without the same runtime costs as `gob`, especially when creating an encoder per call. This can be left for a followup PR though.

We can likely do this by creating protobuf definitions for our `Series` and `ChunkRef` types, which will make this even easier to use over the wire via `proto.Marshal()` and allow our index-gateway to expose these methods via grpc (cc @cyriltovena we talked about this recently).

Reply:
@owen-d can you link to something or provide a bit more background on the runtime cost of `gob`?

Reply:
careful with protobuf it sux for arrays, it's a good idea to keep the size of the array somewhere to speed up unmarshalling.